lib/reek/smells/smell_warning.rb

```diff
@@ -88,25 +95,23 @@ def smell_classes
         [smell_detector.smell_category, smell_detector.smell_type]
       end
 
-      def common_parameters_equal?(other_parameters)
-        other_keys   = other_parameters.keys
-        other_values = other_parameters.values
+      def check_attributes_comparability(other_attributes)
+        other_attributes.keys.each do |attr|
+          unless COMPARABLE_ATTRIBUTES.include?(attr)
+            raise ArgumentError, "The attribute #{attr} you want to check for doesn't exist"
+          end
+        end
+      end
 
+      def common_parameters_equal?(other_parameters)
+        return true unless other_parameters
```

Comment:
I get errors if I remove this.  We have situations where we are not defining any parameters to match against - note how we are calling it:

```ruby
      def matches_attributes?(other_attributes = {})
        check_attributes_comparability(other_attributes)
        return false unless common_parameters_equal?(other_attributes[:parameters])
```

so if other_attributes has no :parameters key (which is often the case) then `common_parameters_equal?` often gets called with a nil which we need to defend against